web/src/views/student/SchedulingScreenStudents.vue

```diff
@@ -121,7 +119,92 @@ import { useAuthStore } from "@/stores/auth";
 import { tryOrAlertAsync } from "@/try";
 
 const snackbar = ref(false);
-const current_id = ref(0);
+const current_round_id = ref(0);
+const current_schedule_id = ref(0);
+const current_progress_id = ref(0);
+
+/**
+ * Use the vue router to navigate to the round set by `setCurrentRound`.
+ */
+function goToRound() {
+  router.push({
+    name: "round_detail",
+    params: { id: current_round_id.value, schedule: current_schedule_id.value },
+  });
+}
+
+/**
+ * Save the id of a round and it's schedule to a temp value
+ * @param schedule
+ */
+function setCurrentRound(schedule: Result<ScheduleQuery>) {
+  current_round_id.value = schedule.round_id;
+  current_schedule_id.value = schedule.id;
+}
+
+/**
+ * Function to open te popup, and save the round details for the popup to be used
+ * @param schedule
+ */
+function openPopup(schedule: {
+  schedule: Result<ScheduleQuery>;
+  progress: Array<Result<ProgressQuery>>;
+}) {
+  current_progress_id.value = schedule.progress[0].id;
+  setCurrentRound(schedule.schedule);
+  snackbar.value = true;
+}
+
+/**
+ * Return whether the start round button should be displayed
+ * @param schedule
+ */
+function showStartButton(schedule: {
+  schedule: Result<ScheduleQuery>;
+  progress: Array<Result<ProgressQuery>>;
+}): boolean {
+  const today = new Date().getDate();
+  const day = new Date(schedule.schedule.day).getDate();
+  return today === day && !roundStarted(schedule.progress);
+}
+
+/**
+ * Check if any building of the round has been started.
+ * @param progresses
+ */
+function roundStarted(progresses: Array<Result<ProgressQuery>>): boolean {
+  for (const progress of progresses) {
+    if (progress.arrival != null) {
+      return true;
+    }
+  }
+  return false;
+}
+
+/**
+ * Count how many buildings are completed for a given round.
+ * @param progresses
+ */
+function getCompletedBuildings(
+  progresses: Array<Result<ProgressQuery>>,
+): number {
+  let count: number = 0;
+  for (const progress of progresses) {
+    if (progress.departure != null) {
+      count++;
+    }
+  }
+  return count;
+}
+
+async function saveStartTime() {
+  await tryOrAlertAsync(async () => {
+    await new ProgressQuery().updateOne({
```

Comment:
Momenteel krijg ik van deze lijn code een QueryError2. Dit is echter iets dat nogal vaak voorkomt en niet aan jouw pagina ligt voor zover ik weet.

Reply:
Dit is hetzelfde probleem als in #270. Een id wordt niet meer door API in een PATCH aanvaard, terwijl de querybuilder dit expliciet wel verwacht. Ik had het hier gisteren met Jozef over. @TheMessik zou jij dit in de API of QueryBuilder kunnen oplossen.